src/shogun/evaluation/MeanSquaredError.cpp

```diff
@@ -16,12 +16,15 @@ float64_t CMeanSquaredError::evaluate(CLabels* predicted, CLabels* ground_truth)
 	REQUIRE(predicted, "Predicted labels must be not null.\n")
 	REQUIRE(ground_truth, "Ground truth labels must be not null.\n")
 	REQUIRE(predicted->get_num_labels() == ground_truth->get_num_labels(), "The number of predicted labels (%d) must be equal to the number of ground truth labels (%d).\n")
-	REQUIRE(predicted->get_label_type() == LT_REGRESSION, "Predicted label type (%d) must be regression (%d).\n", predicted->get_label_type(), LT_REGRESSION)
-	REQUIRE(ground_truth->get_label_type() == LT_REGRESSION, "Ground truth label type (%d) must be regression (%d).\n", ground_truth->get_label_type(), LT_REGRESSION)
 	int32_t length = predicted->get_num_labels();
 	float64_t mse = 0.0;
+	auto predicted_regression = regression_labels(predicted);
+	auto ground_truth_regression = regression_labels(ground_truth);
+
 	for (int32_t i=0; i<length; i++)
-		mse += CMath::sq(((CRegressionLabels*) predicted)->get_label(i) - ((CRegressionLabels*) ground_truth)->get_label(i));
+		mse += CMath::sq(
```

Comment:
`std::`?

Reply:
this is autoformater, I didnt change that line

Reply:
ah no actually I did
sure can change that

Reply:
doesn't matter based on your changes already going to create a problem from the other PR....